docs/source/CUSTOM-JOB.rst

```diff
@@ -0,0 +1,435 @@
+
+Currently, `deepcell.org <www.deepcell.org>`_ supports a cell tracking feature which is facilitated by the ``tracking-consumer``, which handles the multi-step process of cell tracking:
+
+  1. Send each frame of the dataset for segmentation
+  2. Retrive model predictions and run post-processing to generate cell segmentation masks
+  3. Send cell segmentation masks for cell tracking predictions
+  4. Compile final tracking results and post for download
+
+New data processing pipelines can be implemented by writing a custom consumer. The model can be exported for tf-serving using :func:`~deepcell.utils.export_utils.export_model`.
+
+The following variables will be used throughout the setup of the custom consumer. Pick out names that are appropriate for your consumer.
+
+.. py:data:: queue_name
+
+    Specifies the queue name that will be used to identify jobs for the :mod:`redis_consumer`, e.g. ``'track'``
+
+.. py:data:: consumer_name
+
+    Name of custom consumer, e.g. ``'tracking-consumer'``
+
+.. py:data:: consumer_type
+
+    Name of consumer job, e.g. ``'tracking'``
+
+
+Designing a custom consumer
+---------------------------
+
+Consumers consume redis events. Each type of redis event is put into a separate queue (e.g. ``predict``, ``track``), and each consumer type will pop items to consume off that queue.
+
+Each redis event should have the following fields:
+
+* ``model_name`` - The name of the model that will be retrieved by TensorFlow Serving from ``gs://<bucket-name>/models``
+* ``model_version`` - The version number of the model in TensorFlow Serving
+* ``input_file_name`` - The path to the data file in a cloud bucket.
+
+.. todo::
+
+    Does the user need to set ``model_name`` and ``model_version`` somewhere?
+
+If the consumer will send data to a TensorFlow Serving model, it should inherit from :class:`redis_consumer.consumers.TensorFlowServingConsumer`, which has methods :meth:`~redis_consumer.consumers.TensorFlowServingConsumer._get_predict_client` and :meth:`~redis_consumer.consumers.TensorFlowServingConsumer.grpc_image` which can send data to the specific model.  The new consumer must also implement the :meth:`~redis_consumer.consumers.TensorFlowServingConsumer._consume` method which performs the bulk of the work. The :meth:`~redis_consumer.consumers.TensorFlowServingConsumer._consume` method will fetch data from redis, download data file from the bucket, process the data with a model, and upload the results to the bucket again. See below for a basic implementation of :meth:`~redis_consumer.consumers.TensorFlowServingConsumer._consume`:
```

Comment:
Seems to me that this documentation should live in `kiosk-redis-consumer`.  If any of these names change in one repo, we have to update this repo manually as well.

Reply:
Relocated into `kiosk-redis-consumer` docs